media_kit/lib/src/player/libmpv/player/real.dart

```diff
@@ -1136,16 +1136,17 @@ class libmpvPlayer extends PlatformPlayer {
 
     if (event.ref.event_id == generated.mpv_event_id.MPV_EVENT_START_FILE) {
       if (isPlayingStateChangeAllowed) {
-        state = state.copyWith(
-          playing: true,
-          completed: false,
-        );
+        state =
+            state.copyWith(playing: true, completed: false, buffering: true);
         if (!playingController.isClosed) {
           playingController.add(true);
         }
         if (!completedController.isClosed) {
           completedController.add(false);
         }
+        if (!bufferingController.isClosed) {
+          bufferingController.add(true);
```

Comment:
@zezo357 Did you test if maybe this change alone is enough?

I mean. From what I checked, when using `paused-for-cache` you still get events for buffering=false once buffering is done. So emitting initial buffering=true after file is starting to load seems to be enough and no need to switch to `core-idle`. Isn't core-idle switching to true also when video is paused (so it would be misleading to consider that buffering)?

Reply:
from my testing paused-for-cache is never true even when buffering 

and yes core-idle is true when video is paused, so i did make the buffering a combination of (idle and playing) so it will be only true if idle for buffering

Reply:
also since you mentioned that, i will check if paused-for-cache can be used, and valid using the current tests

Reply:
As you can see, the tests fails 

And seek doesn't buffer even though it should

Reply:
Weird behavior on mpv side, but nice workaround for it. Thanks for checking!

Reply:
Thank you for reviewing the PR.

Reply:
The `pause-for-cache` gets `true` (as intended) e.g. internet gets slow/disconnected in the middle of playback.

The only issue with it was that it never worked during "starting" i.e. initial buffering of the stream. That's one thing we're looking for.

Reply:
> The `pause-for-cache` gets `true` (as intended) e.g. internet gets slow/disconnected in the middle of playback... and position crosses the buffered zone.
> 
> The only issue with it was that it never worked during "starting" i.e. initial buffering of the stream. That's one thing we're looking for.

well as you can see in this https://github.com/zezo357/media_kit/actions/runs/5356947170 it never goes true, same for the issue https://github.com/alexmercerind/media_kit/issues/227

Reply:
It goes true, we're using it... only when it pauses for buffering, not at the start of stream.

Reply:
> It goes true, we're using it... only when it pauses for buffering, not at the start of stream.

well that didnt happen in the test mentioned and in the issue

Reply:
I've built this; I know better :)

Reply:
> 

want me to switch back to `pause-for-cache` ? and change logic back to it? even though that from my tests/ usage of the package the buffering was never true in my case (it will pause to buffer but the buffering is not true) keep in mind this is for local files too 

like after seek it will pause for a small bit without buffering `true` in this pr the buffering will be true